advanced/sr_gui_fingertip_visualization/src/sr_gui_fingertip_visualization/visual_tab_layouts.py

```diff
@@ -0,0 +1,439 @@
+                    self._data_labels[finger]['pac1'].setText("pac1:{}".format(self._data[finger]['pac1']))
+                    self._data_labels[finger]['pdc'].setText("pdc:{}".format(self._data[finger]['pdc']))
+                    self._data_labels[finger]['tac'].setText("tac:{}".format(self._data[finger]['tac']))
+                    self._data_labels[finger]['tdc'].setText("tdc:{}".format(self._data[finger]['tdc']))
+
+                elif type(self._finger_widgets[finger]) == DotUnitBiotacSPMinus:
+                    self._finger_widgets[finger].update_data(self._data[finger])
+
+
+class VisualizationTab(QWidget):
+    def __init__(self, parent, tactile_topics):
+        super().__init__(parent=parent)
+        self._tactile_topics = tactile_topics
+        self._init_layout()
+        self._create_connections()
+
+    def _init_layout(self):
+
+        self.finger_layout = QVBoxLayout(self)
+
+        self.options_layout_groupbbox = QGroupBox("Options")
+        self.options_layout_groupbbox.setSizePolicy(1, 2)
+        self.options_layout = QHBoxLayout()
+
+        self.hand_id_selection_layout = QFormLayout()
+        self.hand_id_selection = QComboBox()
+        self.hand_id_selection.addItems(self._tactile_topics.keys())
+        self.hand_id_selection_layout.addRow(QLabel("Hand ID:"), self.hand_id_selection)
+
+        self.data_type_selection_button = QPushButton("Show pac")
+
+        self.finger_selection_label = QLabel("Finger selection:")
+        self.finger_selection_show_selected_button = QPushButton("Show selected")
+        self.finger_selection_show_selected_button.setSizePolicy(2, 2)
+        self.finger_selection_show_all_button = QPushButton("Show all")
+        self.finger_selection_show_all_button.setSizePolicy(2, 2)
+
+        self.options_layout.addLayout(self.hand_id_selection_layout)
+        self.options_layout.addStretch(1)
+        self.options_layout.addWidget(self.data_type_selection_button)
+        self.options_layout.addWidget(self.finger_selection_label)
+        self.options_layout.addWidget(self.finger_selection_show_selected_button)
+        self.options_layout.addWidget(self.finger_selection_show_all_button)
+
+        self.options_layout_groupbbox.setLayout(self.options_layout)
+        self.stacked_layout = QStackedLayout(self)
+
+        self.fingertip_widget = dict()
+        for side, tactile_topic in self._tactile_topics.items():
+            if tactile_topic == "ShadowPST":
+                self.fingertip_widget[side] = PSTVisualizationTab(side, self)
+            elif tactile_topic == "BiotacAll":
+                self.fingertip_widget[side] = BiotacVisualizationTab(side, self)
+            self.stacked_layout.addWidget(self.fingertip_widget[side])
+
+        self.finger_layout.addWidget(self.options_layout_groupbbox)
+        self.finger_layout.addLayout(self.stacked_layout)
+        self.setLayout(self.finger_layout)
+
+        self._current_side = list(self._tactile_topics.keys())[0]
```

Comment:
Whit not just using next(self._tactile_topics.keys())? Way cleaner and avoid unnecessary casting